public/js/controllers/charts_controller.js

```diff
@@ -434,66 +453,69 @@ export default class extends Controller {
     this.lastZoom = Zoom.object(start, end)
     this.settings.zoom = Zoom.encode(this.lastZoom)
     this.query.replace(this.settings)
-    this.setSelectedZoom(Zoom.mapKey(this.lastZoom, this.limits))
   }
 
-  setSelectedZoom (zoomKey) {
-    this.zoomOptionTargets.forEach((button) => {
-      if (button.dataset.zoom === zoomKey) {
-        button.classList.add('active')
-      } else {
-        button.classList.remove('active')
-      }
-    })
+  setZoom (e) {
+    var target = e.srcElement || e.target
+    var option = target ? target.dataset.option : e
+    if (!option) return
+    this.setActiveOptionBtn(option, this.zoomOptionTargets)
+    if (option === e) return // Exit if running for the first time.
+    this.validateZoom()
   }
 
   setBin (e) {
     var target = e.srcElement || e.target
-    this.binSizeTargets.forEach(li => li.classList.remove('active'))
-    target.classList.add('active')
-    userBins[selectedChart] = target.dataset.bin
+    var option = target ? target.dataset.option : e
+    if (!option) return
+    this.setActiveOptionBtn(option, this.binSizeTargets)
+    if (option === e) return // Exit if running for the first time.
     selectedChart = null // Force fetch
     this.selectChart()
   }
 
-  setBinButton (bin) {
-    this.binSizeTargets.forEach(li => {
-      if (li.dataset.bin === bin) {
```

Comment:
Here too, `setBinButton` sets the selected button, while `setBin` changes `this.settings.bin` and causes a chart refresh, which is not desirable during initialization.